File: jscomp/test/gpr_4280_test.js
'use strict';

var Mt = require("./mt.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  return Mt.eq_suites(test_id, suites, loc, x, y);
}

var u = {
  contents: 0
};

function div(children, param) {
  for(var i = 0; i <= 1; ++i){
    u.contents = 300;
    console.log("nonline");
  }
  
}

function string(s) {
  for(var i = 0; i <= 1; ++i){
    u.contents = 200;
    console.log("no");
  }
  
}

function fn(authState, route) {
  var onboardingRoute;
  if (typeof authState === "string") {
    var exit = 0;
    if (typeof route === "string") {
      if (route === "SignUp" || route === "SignIn" || route === "Invite" || route === "PasswordReset") {
        div({
              hd: string("LoggedOut"),
              tl: /* [] */0
            }, undefined);
        return 1;
      }
      exit = 2;
    } else if (route.HASH === "Onboarding") {
      onboardingRoute = route.VAL;
    } else {
      exit = 2;
    }
    if (exit === 2) {
      div({
            hd: string("Redirect"),
            tl: /* [] */0
          }, undefined);
      return 3;
    }
    
  } else {
    var exit$1 = 0;
    if (typeof route === "string" || route.HASH !== "Onboarding") {
      exit$1 = 2;
    } else {
      onboardingRoute = route.VAL;
    }
    if (exit$1 === 2) {
      console.log(authState.VAL);
      div({
            hd: string("VerifyEmail"),
            tl: /* [] */0
          }, undefined);
      return 2;
    }
    
  }
  console.log(onboardingRoute);
  div({
        hd: string("Onboarding"),
        tl: /* [] */0
      }, undefined);
  return 0;
}

eq("File \"gpr_4280_test.ml\", line 46, characters 6-13", fn("Unauthenticated", "Invite"), 1);

eq("File \"gpr_4280_test.ml\", line 47, characters 6-13", fn("Unauthenticated", {
          HASH: "Onboarding",
          VAL: 0
        }), 0);

eq("File \"gpr_4280_test.ml\", line 48, characters 6-13", fn({
          HASH: "Unverified",
          VAL: 0
        }, "Invite"), 2);

eq("File \"gpr_4280_test.ml\", line 49, characters 6-13", fn("Unauthenticated", "xx"), 3);

Mt.from_pair_suites("gpr_4280_test.ml", suites.contents);

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.u = u;
exports.div = div;
exports.string = string;
exports.fn = fn;
/*  Not a pure module */

File: jscomp/test/gpr_4280_test.ml
[@@@bs.config {flags = [|(* "-bs-diagnose" *)|]}]
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y = Mt.eq_suites ~test_id ~suites loc x y 

let u = ref 0
let div ~children () = 
  for i = 0 to 1 do   
    u := 300;  
    Js.log "nonline"
  done

let string (s : string) =   
  for i = 0 to 1 do 
    u := 200;
    Js.log "no"
  done     

let fn authState route =
  match (authState, route) with
  | (`Unauthenticated,`Onboarding onboardingRoute)
  | (`Unverified _,`Onboarding onboardingRoute) ->
    Js.Console.log onboardingRoute;
    div
      ~children:[string
                   "Onboarding"] ();
      0
  | (`Unauthenticated,`SignIn)|(`Unauthenticated,`SignUp)
  |(`Unauthenticated,`Invite)|(`Unauthenticated,`PasswordReset) ->
    div
      ~children:[string
                   "LoggedOut"] ();1

  | (`Unverified user,_) ->
    Js.Console.log user;
    div
      ~children:[string
                   "VerifyEmail"] ();
      2
  | (`Unauthenticated,_) ->
    div
      ~children:[string
                   "Redirect"] ();3


;; eq __LOC__ (fn (`Unauthenticated) `Invite) 1
;; eq __LOC__ (fn (`Unauthenticated) (`Onboarding 0)) 0
;; eq __LOC__ (fn (`Unverified 0) `Invite) 2 
;; eq __LOC__ (fn `Unauthenticated `xx) 3 
;; Mt.from_pair_suites __FILE__ !suites
